### PR TITLE
Feature branch to Development branch

### PR DESCRIPTION
**Pull request by daviidy**
We had to build a site similar to a private Eventbrite which allows users to create events and then manage user signups. Users can create events and send invitations and parties (sound familiar?). Events take place at a specific date and at a location (which you can just store as a string, like “Andy’s House”).

A user can create events. A user can attend many events. An event can be attended by many users.
Users have to be authenticated before creating an event.
We added validations and associations tests for User, Event, and Attendance models.
We added also integration tests for authentication and event creation

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  has_many :invited_users, through: :attendances
+  validates_presence_of :location, :description, :title, :date
+
+  # def self.past
+  #   self.where("date < ?", Date.today)
+  # end
+  #
+  # def self.upcoming
+  #   self.where("date > ?", Date.today)
+  # end
```
Comment (BrittanyBlake) [inline app/models/event.rb]: - [x] Remove commented code. If you need to see a previous version you can refer to previous commits.

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+class EventsController < ApplicationController
+  before_action :authorize, except: %i[index show]
+
+  def index
+    # @events = Event.all.order('created_at DESC')
+    # @past = Event.past.order("date DESC")
+    # @upcoming = Event.upcoming.order("date DESC")
```
Comment (BrittanyBlake) [inline app/controllers/events_controller.rb]: - [x] Please remove commented code.

File: app/views/events/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                  </div>
+                  <div class="event-title">
+                    <h3 class="font-weight-bold">
+                      <%= link_to event.title, event_path(event), class: "text-dark" %>
+                    </h3>
+                  </div>
+                </div>
+              </div>
+            <% end %>
+
```
Comment (BrittanyBlake) [inline app/views/events/index.html.erb:47]: - [x] You should be displaying previously attended events by using event model scope in the show action of your User's controller. 
Screen Shot 2020-12-10 at 8 28 56 PM

File: app/views/events/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                    <span class="text-uppercase text-danger"><%= event.date %></span>
+                  </div>
+                  <div class="event-title">
+                    <h3 class="font-weight-bold">
+                      <%= link_to event.title, event_path(event), class: "text-dark" %>
+                    </h3>
+                  </div>
+                </div>
+              </div>
+            <% end %>
```
Comment (BrittanyBlake) [inline app/views/events/index.html.erb:66]: - [x] You should be displaying upcoming events by using event model scope in the show action of your User's controller. 
Screen Shot 2020-12-10 at 8 28 56 PM

**Review by BrittanyBlake — CHANGES_REQUESTED** [review]
### Status: Changes Required ♻️ 
****
### Hi @daviidy and @AkintoyeOlamide! 👋 
You have done a really awesome job with this project!!👍 It looks amazing 🤩 🤩 
There are just a few issues that need to be resolved before you can move on: ⬇️ 
****

### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**